core/aws-core/src/main/java/software/amazon/awssdk/awscore/client/builder/AwsDefaultClientBuilder.java

```diff
@@ -374,7 +377,13 @@ private RetryPolicy resolveAwsRetryPolicy(SdkClientConfiguration config) {
 
     @Override
     public final BuilderT region(Region region) {
-        clientConfiguration.option(AwsClientOption.AWS_REGION, region);
+        if (idHasFipsPattern(region)) {
+            clientConfiguration.option(AwsClientOption.AWS_REGION, removeFips(region));
+            clientConfiguration.option(AwsClientOption.FIPS_ENDPOINT_ENABLED, true);
+        } else {
+            clientConfiguration.option(AwsClientOption.AWS_REGION, region);
+            clientConfiguration.option(AwsClientOption.FIPS_ENDPOINT_ENABLED, null);
+        }
```

Comment:
Performance nit: can we do the starts-with and ends-with check once? Can we also do substring to remove the prefix/suffix instead of replaceOnce (which can behave weird for things like foo-fips-1-fips)?

Reply:
Weird, good callout on the behavior. Will fix. Will fix avoiding the multiple calls to starts/endsWith.

Reply:
Fixed. I optimized for the prefix/suffix case since that's likely to be the majority anyway. In any case, this only happens during client build time.